patchers/value/value.go

```diff
@@ -0,0 +1,260 @@
+// Package value provides a Patcher that uses interfaces to allow custom types that can be represented as standard go values to be used more easily in expressions.
+//
+// # Example Usage
+//
+//	import (
+//		"fmt"
+//		"github.com/antonmedv/expr/patchers/value"
+//		"github.com/antonmedv/expr"
+//	)
+//
+//	type customInt struct {
+//	       	Int int
+//	}
+//
+//	// Provides type checking at compile time
+//	func (v *customInt) IntValue() int {
+//	       	return v.Int
+//	}
+//
+//	// Lets us return nil if we need to
+//	func (v *customInt) ExprValue() any {
+//	       	return v.Int
+//	}
+//
+//	func main() {
+//		env := make(map[string]any)
+//		env["ValueOne"] = &customInt{1}
+//		env["ValueTwo"] = &customInt{2}
+//
+//		program, err := expr.Compile("ValueOne + ValueTwo", expr.Env(env), value.Patcher)
+//
+//		if err != nil {
+//			panic(err)
+//		}
+//
+//		out, err := vm.Run(program, env)
+//
+//		if err != nil {
+//			panic(err)
+//		}
+//
+//		fmt.Printf("Got %v", out)
+//	}
+package value
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/antonmedv/expr"
+	"github.com/antonmedv/expr/ast"
+	"github.com/antonmedv/expr/conf"
+)
+
+// Patcher is an expr.Option that both patches the program and adds the `getExprValue` function.
+// Use it directly as an Option to expr.Compile()
+var Patcher = func() expr.Option {
```

Comment:
Let's use patcher pkg and rename Patcher to, for example, `patcher.ValueGetter`.

Reply:
How should I define the actual expr.Function doing this? Should I move it somewhere else?

`expr` imports `patcher` so it creates an import cycle to create the function in `patcher`.

Reply:
@antonmedv

Any advice on where I should define the expr.Function without causing an import cycle?

I don't think it should be defined as a normal builtin function?

Reply:
Sorry, I didn't understand the question. 

I mean this: lets move code to patchers dir and rename this function to ValueGetter.